packages/client/hooks/useTrebuchetEvents.ts

```diff
@@ -55,16 +55,34 @@ const useTrebuchetEvents = () => {
 
     const onReconnected = () => {
       setConnectedStatus(true)
-      atmosphere.retries.forEach((retry) => retry())
```

Comment:
Looks like there is no way to know the new version on reconnect as it is not possible to read handshake headers. 
The only way to manage reconnect logic based on a version is to retrieve the version from the backend. Backend is already sending the new version automatically on receiving a connection. So I tried to move this logic in `onData` below.